CSharp/JobPrep/Program.cs

```
@@ -0,0 +1,333 @@
+            using (IBatchClient client = BatchClient.Connect(Url, cred))
+            {
+                // if you want to put a retry policy in place, enable it here
+                // the built-in policies are No Retry (default), Linear Retry, and Exponential Retry
+                //client.CustomBehaviors.Add(new SetRetryPolicy(new Microsoft.Azure.Batch.Protocol.LinearRetry()));
+
+                ListPools(client);
+                ListWorkItems(client);
+
+                CreatePoolIfNotExist(client, Program.PoolName);
+                AddWork(client);
+
+                ListPools(client);
+                ListWorkItems(client);
+            }
+
+            Console.WriteLine("Press return to exit...");
+            Console.ReadLine();
+        }
+
+        private static void CreatePoolIfNotExist(IBatchClient client, string poolName)
+        {
+            // All Pool and VM operation starts from PoolManager
+            using (IPoolManager pm = client.OpenPoolManager())
+            {
+                // go through all the pools and see if it already exists
+                bool found = false;
+                foreach (ICloudPool p in pm.ListPools())
+                {
+                    // pools are uniquely identified by their name
+                    if (string.Equals(p.Name, poolName))
+                    {
+                        Console.WriteLine("Using existing pool {0}", poolName);
+                        found = true;
+
+                        if (!p.ListVMs().Any<IVM>())
+                        {
+                            Console.WriteLine("There are no VMs in this pool. No tasks will be run until at least one VM has been added via resizing.");
+                            Console.WriteLine("Resizing pool to add 3 VMs. This might take a while...");
+                            p.Resize(3);
+                        }
+
+                        break;
+                    }
+                }
+
```

This is maybe a minor issue, but we shouldn't be recommending customers list all their pools and go through them in order to see if their pool already exists.  We can just do a GetPool and swallow a NotFound exception.  It's more efficient that way.